lib/src/room.dart

```diff
@@ -281,32 +304,39 @@ class Room {
     return t is String ? t : '';
   }
 
-  /// The avatar of the room if set by a participant.
+  /// The avatar of the room if set by a participant. If you want to also
+  /// let the avatar be calculated by the room heroes for DM rooms, use
+  /// `Room.calcAvatar()` instead.
   Uri? get avatar {
-    final avatarUrl = getState(EventTypes.RoomAvatar)?.content['url'];
-    if (avatarUrl is String) {
-      return Uri.tryParse(avatarUrl);
-    }
+    final avatarUrl =
+        getState(EventTypes.RoomAvatar)?.content.tryGet<String>('url');
+    if (avatarUrl != null) return Uri.tryParse(avatarUrl);
 
-    final heroes = summary.mHeroes;
-    if (heroes != null && heroes.length == 1) {
-      final hero = getState(EventTypes.RoomMember, heroes.first);
-      if (hero != null) {
-        return hero.asUser.avatarUrl;
-      }
-    }
-    if (isDirectChat) {
-      final user = directChatMatrixID;
-      if (user != null) {
-        return unsafeGetUserFromMemoryOrFallback(user).avatarUrl;
+    return null;
+  }
+
+  /// Calculates the avatar. First checks if `m.room.avatar` is set and then
+  /// tries to calculate it from the room heroes. It will use the first
+  /// hero whos avatar is not null.
+  /// For invitation this displays the avatar of the invitor as fallback.
```

Comment:
Do you want to reference https://spec.matrix.org/v1.9/client-server-api/#calculating-the-display-name-for-a-room here and why we diverge from it?

Reply:
We should document if we differ from the spec on purpose. We don't do this on purpose here as we have implemented this logic a long time ago. I have tried do not change the logic much here (I think I improved it a little bit by not using the first user from heroes but the first one with an avatar url, but thats not a big change IMO). Changing it so that it is according to the spec could be a different issue